source/frontend/src/components/maps/leaflet/mapUtils.tsx

```diff
@@ -261,28 +261,20 @@ export const toCqlFilter = (
  * NOTE: The resulting string value is left as-is. It will not be URL-encoded
  * @param object an object to convert to a cql filter string.
  */
-export const toCqlFilterValue = (
-  object: Record<string, string>,
-  forceSimplePid?: boolean,
-  forceExactMatch?: boolean,
-) => {
+export const toCqlFilterValue = (object: Record<string, string>, flags?: IWfsCqlFlags) => {
   const cql: string[] = [];
   Object.keys(object).forEach((key: string) => {
     if (object[key]) {
       if (
         ((key === 'PID' || key === 'PID_PADDED') && object[key]?.length === 9) ||
-        forceExactMatch
+        flags?.forceExactMatch
       ) {
         cql.push(`${key} = '${object[key]}'`);
-      } else if ((key === 'PID' || key === 'PID_PADDED') && object[key] && !forceSimplePid) {
```

Comment:
if code needs to filter by pid and pid_padded and pin it should do that by providing those three keys in the object passed to this function not by falling into this specific case.

Reply:
Thanks for fixing this logic. I never liked this built-in edge case